### PR TITLE
Allow theming of header edit background color

### PR DESCRIPTION
**Pull request by KTibow**
## Proposed change

This lets themes apply their own background coloring to the header during edit mode, using the variable `--app-header-edit-background-color`.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
frontend:
  themes:
    test:
      app-header-edit-background-color: red
```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: Historical #7480
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/hui-root.ts [hunk 1]
```diff
@@ -815,7 +815,7 @@ class HUIRoot extends LitElement {
 
         .edit-mode app-header,
         .edit-mode app-toolbar {
-          background-color: var(--dark-color, #455a64);
+          background-color: var(--app-header-edit-background-color, var(--dark-color, #455a64));
```
Comment (bramkragten) [inline src/panels/lovelace/hui-root.ts:818]: The `--dark-color` variable is only used here, we don't need yet another variable?

**Comment by KTibow** [comment]
Are you suggesting renaming it, or letting users style it? Right now any user styling is overridden by a :host style somewhere.

File: src/panels/lovelace/hui-root.ts (hunk 1 above)
Comment (bramkragten) [reply src/panels/lovelace/hui-root.ts:818]: I suggest you rename `dark-color` to `--app-header-edit-background-color`, and the same for the text color.

This should be removed:
https://github.com/home-assistant/frontend/blob/zwave-ozw-migration/src/panels/lovelace/hui-root.ts#L787-L788

File: src/panels/lovelace/hui-root.ts [hunk 2]
```diff
@@ -815,7 +815,7 @@ class HUIRoot extends LitElement {
...
           color: var(--text-dark-color);
```
Comment (KTibow) [inline src/panels/lovelace/hui-root.ts:819]:
```suggestion
          background-color: var(--app-header-edit-background-color, #455a64);
          color: var(--app-header-text-color);
```

File: src/panels/lovelace/hui-root.ts [hunk 3]
```diff
@@ -815,8 +813,8 @@ class HUIRoot extends LitElement {
 
         .edit-mode app-header,
         .edit-mode app-toolbar {
-          background-color: var(--dark-color, #455a64);
-          color: var(--text-dark-color);
+          background-color: var(--app-header-edit-background-color, #455a64);
+          color: var(--app-header-text-color, #fff);
```
Comment (bramkragten) [inline src/panels/lovelace/hui-root.ts:817]:
```suggestion
          color: var(--app-header-edit-text-color, #fff);
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**